mvndz/src/main/java/com/gd/internship/me/DZ02Streams.java

```diff
@@ -0,0 +1,115 @@
+package com.gd.internship.me;
+import com.gd.internship.me.User;
+import com.github.javafaker.Faker;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class DZ02Streams {
+    static long id  = 0;
+
+    public static void main(String[] args) {
+        Map<Long, User> idToUser = new HashMap<>();
+
+
+        setMapData(idToUser);
+
+        // if you need see all map
+//        for (Map.Entry<Long, User> pair : users.entrySet()){
+//            System.out.println(pair.getKey() + " : " + pair.getValue().getName());
+//        }
+
+        List<User> collectId = idToUser.entrySet().stream()
+                .filter(pair -> pair.getKey() <= 121).map(x -> x.getValue())
+                .collect(Collectors.toList());
+
+        List<String> collectAge = idToUser.entrySet().stream()
+                .filter(pair -> pair.getValue().getAge() > 18)
+                .map(x -> x.getValue().getSurname())
+                .collect(Collectors.toList());
+
+        List<String> collectSurnamesA = idToUser.entrySet().stream()
+                .filter(pair -> pair.getValue().getSurname().toCharArray()[0] == 'A')
```

Comment:
`pair.getValue().getSurname().toCharArray()[0] == 'A'` - when you compare chars or strings like this it is better to use

`'A' == pair.getValue().getSurname().toCharArray()[0]` 

form - it is morr Null-safer.

In case of strings - use `#equals()`